acapy_agent/wallet/askar.py

```
@@ -878,7 +890,7 @@ def _create_keypair(key_type: KeyType, seed: Union[str, bytes, None] = None) ->
         raise WalletError(f"Unsupported key algorithm: {key_type}")
     if seed:
         try:
-            if key_type == ED25519:
+            if key_type == ED25519 or key_type == P256:
                 # not a seed - it is the secret key
                 seed = validate_seed(seed)
                 return Key.from_secret_bytes(alg, seed)
```

Not part of your changes, but I notice in lines 899--901:
```py
        except AskarError as err:
            if err.code == AskarErrorCode.INPUT:
                raise WalletError("Invalid seed for key generation") from None
```
This will swallow AskarError's that don't match that error code, and silently fail. i.e. any other error code will go unlogged / unreported.
And I'm not sure why it's `from None` and not `from err`.

Could we perhaps, while we're here, change that to:
```py
        except AskarError as err:
            if err.code == AskarErrorCode.INPUT:
                raise WalletError("Invalid seed for key generation") from err
            else:
                logger.error(f"Unhandled Askar error code: {err.code}")
                raise
```

---

good find - added that in